compiler/x/codegen/X86BinaryEncoding.cpp

```diff
@@ -948,6 +948,21 @@ TR::X86ImmInstruction::addMetaDataForCodeAddress(uint8_t *cursor)
                   (uint8_t *)(intptr_t)getNode()->getInlinedSiteIndex() , (TR_ExternalRelocationTargetKind) getReloKind(), cg()),
                   __FILE__, __LINE__, getNode());
                break;
+            case TR_MethodPointer:
+               if (getNode() && getNode()->getInlinedSiteIndex() == -1)
```

Comment:
I think you should include in this condition a test that the immediate value matches the J9Method of the compiled method. Otherwise, any instruction generated from a node in the method being compiled that has a MethodPointer will be replaced with the compiled method's J9Method (because that's what RamMethod does). Although that *probably* works in all existing cases, I don't think it's necessarily the right thing to do. Only if the method pointer is the current method's J9Method is it definitely the right thing to do.